api/v2/serializers/details/volume.py

```diff
@@ -55,50 +51,15 @@ class Meta:
             'projects',
             'size',
             'url',
-
-            'snapshot_id',
-            'image_id',
-
             'start_date',
             'end_date')
 
     def validate(self, data):
-        image_id = data.get('image')
-        snapshot_id = data.get('snapshot')
-
-        #: Only allow one at a time
-        if snapshot_id and image_id:
-            raise serializers.ValidationError(
-                "Use either `snapshot_id` or `image_id` not both.")
-        return data
-
-    def create(self, validated_data):
-        name = validated_data.get('name')
-        size = validated_data.get('size')
-        identifier = validated_data.get("identifier")
-        description = validated_data.get('description')
-        user = validated_data.get("user")
-        start_date = validated_data.get("created_on")
-
-        instance_source = validated_data.get("instance_source")
-        identity = instance_source.get("created_by_identity")
-        provider = identity.provider
-
-        source = InstanceSource.objects.create(
-            identifier=identifier,
-            provider=provider,
-            created_by=user,
-            created_by_identity=identity)
-
-        kwargs = {
-            "name": name,
-            "size": size,
-            "description": description,
-            "instance_source": source,
-            "start_date": start_date
-        }
-
-        return Volume.objects.create(**kwargs)
+        if not data and not self.initial_data:
+            return data
+        raise Exception(
+            "This serializer for GET output ONLY! -- "
+            "Use the POST or UPDATE serializers instead!")
```

Comment:
This block is a little confusing to me. Idk the details of `Serializers` shouldn't we just raise exception if this method is called. I.e Why would validate ever get called w/o `data` seems hacky.